diff --git a/moptions.cpp b/moptions.cpp
index ebc38ea..fc2af3f 100644
--- a/moptions.cpp
+++ b/moptions.cpp
@@ -53,7 +53,10 @@
 	// UI options
 	{ nullptr,                              nullptr,            option_type::HEADER,     "UI OPTIONS" },
 	{ OPTION_INFOS_SIZE "(0.20-1.00)",      "0.75",             option_type::FLOAT,      "UI right panel infos text size (0.20 - 1.00)" },
-	{ OPTION_FONT_ROWS "(25-40)",           "30",               option_type::INTEGER,    "UI font lines per screen (25 - 40)" },
+// DAV HACK	
+	// { OPTION_FONT_ROWS "(25-40)",           "30",               option_type::INTEGER,    "UI font lines per screen (25 - 40)" },
+	{ OPTION_FONT_ROWS "(15-40)",           "20",           option_type::INTEGER,     "UI font lines per screen (15 - 40)" },	
+// END DAV HACK	
 	{ OPTION_UI_BORDER_COLOR,               "ffffffff",         option_type::STRING,     "UI border color (ARGB)" },
 	{ OPTION_UI_BACKGROUND_COLOR,           "ef101030",         option_type::STRING,     "UI background color (ARGB)" },
 	{ OPTION_UI_CLONE_COLOR,                "ff808080",         option_type::STRING,     "UI clone color (ARGB)" },
